src/App.tsx

```diff
@@ -34,62 +35,123 @@ export function getReorderedGoods(
+
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean;
+  sortType: SortType;
+};
+
+// export const App: React.FC = () => {
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleSortAlphClick = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  handleSortLengthClick = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  handleReverseClick = () => {
+    this.setState(state => ({ isReversed: !state.isReversed }));
+  };
+
+  handleResetClick = () => {
+    this.setState({ sortType: SortType.NONE, isReversed: false });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const visibleGoods = getReorderedGoods(
+      goodsFromServer, { sortType, isReversed },
+    );
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button', 'is-info',
+              {
+                'is-light': !(sortType === SortType.ALPHABET),
+              })}
+            // className="button is-info is-light"
+            onClick={this.handleSortAlphClick}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button', 'is-success',
+              {
+                'is-light': !(sortType === SortType.LENGTH),
+              })}
+            // className="button is-success is-light"
+            onClick={this.handleSortLengthClick}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button', 'is-warning',
+              {
+                'is-light': !isReversed,
+              })}
+            // className="button is-warning is-light"
+            onClick={this.handleReverseClick}
+          >
+            Reverse
+          </button>
+
+          {isReversed || sortType !== SortType.NONE
+            ? (
```

Comment:
move to the separate variable
also, use `&&` operator for it
```suggestion
          {... && (
```